cmd/cartesi-rollups-evm-reader/main.go

```diff
@@ -0,0 +1,193 @@
+		"default-block",
+		"d",
+		"latest",
+		`Default block to be used when fetching new blocks.
+		One of 'latest', 'safe', 'pending', 'finalized'`)
+
+	Cmd.Flags().StringVarP(&config.postgresEndpoint,
+		"postgres-endpoint",
+		"p",
+		"postgres://postgres:password@localhost:5432/postgres",
+		"Postgres endpoint")
+
+	Cmd.Flags().StringVarP(&config.blockchainHttpEndpoint,
+		"blockchain-http-endpoint",
+		"b",
+		"http://localhost:8545",
+		"Blockchain HTTP Endpoint")
+
+	Cmd.Flags().StringVarP(&config.blockchainWsEndpoint,
+		"blockchain-ws-endpoint",
+		"w",
+		"ws://localhost:8545",
+		"Blockchain WS Endpoint")
+
+	Cmd.Flags().StringVarP(&config.inputBoxAddress,
+		"inputbox-address",
+		"i",
+		"0xA1b8EB1F13d8D5Db976a653BbDF8972cfD14691C",
+		"Input Box contract address")
+
+	Cmd.Flags().Uint64VarP(&config.inputBoxDeploymentBlockNumber,
+		"inputbox-block-number",
+		"n",
+		devnetInputBoxDeploymentBlockNumber,
+		"Input Box deployment block number")
+
+	Cmd.Flags().BoolVarP(&verbose,
+		"verbose",
+		"v",
+		false,
+		"enable verbose logging")
+}
+
+func main() {
+	err := Cmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func run(cmd *cobra.Command, args []string) {
+	startTime := time.Now()
+
+	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer stop()
+
+	// setup log
+	logLevel := slog.LevelInfo
+	if verbose {
+		logLevel = slog.LevelDebug
+	}
+
+	opts := &tint.Options{
+		Level:      logLevel,
+		AddSource:  logLevel == slog.LevelDebug,
+		TimeFormat: "2006-01-02T15:04:05.000", // RFC3339 with milliseconds and without timezone
+	}
+	handler := tint.NewHandler(os.Stdout, opts)
+	logger := slog.New(handler)
+	slog.SetDefault(logger)
+	slog.Info(
+		"Starting the Cartesi Rollups Node's EVM Reader",
+		"version",
+		buildVersion,
+		"config",
+		config,
+	)
+
+	repository.RunMigrations(fmt.Sprintf("%v?sslmode=disable", config.postgresEndpoint))
+
+	database, err := repository.Connect(ctx, config.postgresEndpoint)
+	if err != nil {
+		slog.Error("EVM Reader exited with an error", "error", err)
+		os.Exit(1)
+	}
+
+	var defaultBlock model.DefaultBlock
+	switch config.defaultBlock {
+	case "latest":
+		defaultBlock = model.DefaultBlockStatusLatest
+	case "safe":
+		defaultBlock = model.DefaultBlockStatusSafe
+	case "pending":
+		defaultBlock = model.DefaultBlockStatusPending
+	case "finalized":
+		defaultBlock = model.DefaultBlockStatusFinalized
+	default:
+		slog.Error("Invalid default block value", "default block", config.defaultBlock)
+		os.Exit(1)
+	}
+
+	// setup database
+	nodePersistentConfig := model.NodePersistentConfig{
+		DefaultBlock:            defaultBlock,
+		InputBoxDeploymentBlock: config.inputBoxDeploymentBlockNumber,
+		InputBoxAddress:         common.HexToAddress(config.inputBoxAddress),
+	}
+	err = database.InsertNodeConfig(ctx, &nodePersistentConfig)
+	if err != nil {
+		slog.Error("EVM Reader exited with an error", "error", err)
+		os.Exit(1)
+	}
+	database.Close()
```

Comment:
why are you closing the connection if this service needs to keep a connection to the database?

Reply:
Now it is a shared resource